packages/devextreme/testing/testcafe/tests/scheduler/layout/templates/appointmentTemplate.ts

```diff
@@ -66,3 +76,274 @@ fixture.disablePageReloads`Layout:Templates:appointmentTemplate`
+      appointmentDates = Array.from(
+        { length: 31 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setDate(result.getDate() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'day':
+    case 'timelineDay':
+    default: {
+      appointmentDates = Array.from(
+        { length: 12 },
+        (_, index) => {
+          const result = new Date(2024, 0, 2, 8);
+          result.setHours(result.getHours() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+  }
+
+  const datesToCheck = [
+    appointmentDates[0],
+    appointmentDates[Math.floor(appointmentDates.length / 3)],
+    appointmentDates[appointmentDates.length - 1],
+  ];
+
+  const groupsToCheck = [
+    { groupId: 0 },
+    { groupId: Math.floor(resourceCount / 3) },
+    { groupId: resourceCount - 1 },
+  ];
+
+  test(`targetedAppointmentData should be correct with groups (viewType="${viewType}", groupOrientation="${groupOrientation}", scrollMode="${scrollMode}", rtlEnabled="${rtlEnabled}") (T1205120)`, async (t) => {
+    const resourceDataSource = Array.from({ length: resourceCount }, (_, index) => ({
+      id: index,
+      text: `Resource ${index}`,
+    }));
+
+    const appointments = resourceDataSource.reduce<Appointment[]>((acc, resource) => acc.concat(
+      appointmentDates
+        .map<Appointment>((date) => ({
+        text: resource.text,
+        startDate: date,
+        endDate: new Date(date.getTime() + HOUR / 2),
+        groupId: resource.id,
+      })),
+    ), []);
+
+    await createWidget('dxScheduler', {
+      rtlEnabled,
+      height: 600,
+      width: 800,
+      currentDate,
+      startDayHour,
+      endDayHour,
+      scrolling: {
+        mode: scrollMode,
+      },
+      groups: ['groupId'],
+      views: [
+        {
+          type: viewType,
+          groupOrientation,
+        },
+      ],
+      currentView: viewType,
+      dataSource: appointments,
+      resources: [
+        {
+          fieldExpr: 'groupId',
+          allowMultiple: true,
+          dataSource: resourceDataSource,
+          label: 'Employees',
+          displayExpr: 'id',
+        },
+      ],
+      appointmentTemplate(model, _, element) {
+        const { groupId: targetedId } = model.targetedAppointmentData;
+        const { groupId } = model.appointmentData;
+
+        if (groupId !== targetedId) {
+          throw new Error('Group ID and targeted ID are mismatched');
+        }
+
+        element.append(`tid[${targetedId}] gid[${groupId}]`);
+        return element;
+      },
+    });
+
+    if (scrollMode === 'virtual') {
+      const scrollOptions = generateOptionMatrix({
+        date: datesToCheck,
+        group: groupsToCheck,
+      });
+
+      // eslint-disable-next-line no-restricted-syntax
+      for (const { date, group } of scrollOptions) {
+        await scrollTo(date, group);
+        await t.wait(50);
+      }
+    }
```

Comment:
Let's reduce complexity of this code block and invert this ```if``` statement:

```suggestion
    if (scrollMode !== 'virtual') {
      return;
    }
    
    const scrollOptions = generateOptionMatrix({
        date: datesToCheck,
        group: groupsToCheck,
      });

      // eslint-disable-next-line no-restricted-syntax
      for (const { date, group } of scrollOptions) {
        await scrollTo(date, group);
        await t.wait(50);
      }
```